lib/data_aggregator/files/attachment/attachment.ex

```diff
@@ -107,5 +116,9 @@ defmodule DataAggregator.Files.Attachment do
   postgres do
     table "file_attachments"
     repo DataAggregator.Repo
+
+    references do
+      reference :collection, on_delete: :nilify, deferrable: true, index?: true
```

Comment:
if the referenced collection is deleted, we just want to empty the field, because we still need the attachment for later async hard deletion of the file stored on s3